examples/xcm/reserve_transfer.eg.ts

```diff
@@ -0,0 +1,224 @@
+          Trappist.XcmV1Junction.PalletInstance((await Statemine.Assets.pallet.run()).id),
+          Trappist.XcmV1Junction.GeneralIndex(BigInt(RESERVE_ASSET_ID)),
+        ),
+      }),
+    }),
+  })
+  .signed(signature({ sender: root }))
+  .sent()
+  .dbgStatus("Trappist(root): Register AssetId to Reserve AssetId")
+  .finalized()
+  .run()
+
+// Reserve transfer asset id on reserve parachain to Trappist parachain.
+{
+  // Destructure the factories to be used (for convenience).
+  const {
+    VersionedMultiLocation,
+    VersionedMultiAssets,
+    Fungibility,
+    XcmV1Junction,
+    Junctions,
+    AssetId,
+    NetworkId,
+    WeightLimit,
+    RuntimeEvent,
+    CumulusPalletXcmpQueueEvent: { isXcmpMessageSent },
+  } = Statemine
+  const events = await Statemine.PolkadotXcm
+    .limitedReserveTransferAssets({
+      dest: VersionedMultiLocation.V1(Rune.rec({
+        parents: 1,
+        interior: Junctions.X1(
+          XcmV1Junction.Parachain(TRAPPIST_CHAIN_ID),
+        ),
+      })),
+      beneficiary: VersionedMultiLocation.V1(Rune.rec({
+        parents: 0,
+        interior: Junctions.X1(
+          XcmV1Junction.AccountId32({
+            network: NetworkId.Any(),
+            id: billy.publicKey,
+          }),
+        ),
+      })),
+      assets: VersionedMultiAssets.V1(Rune.array([Rune.rec({
+        id: AssetId.Concrete(Rune.rec({
+          parents: 0,
+          interior: Junctions.X2(
+            XcmV1Junction.PalletInstance((await Statemine.Assets.pallet.run()).id),
+            XcmV1Junction.GeneralIndex(BigInt(RESERVE_ASSET_ID)),
+          ),
+        })),
+        fun: Fungibility.Fungible(10000000000000n),
+      })])),
+      feeAssetItem: 0,
+      weightLimit: WeightLimit.Unlimited(),
+    })
+    .signed(signature({ sender: billy }))
+    .sent()
+    .dbgStatus("Statemine(Billy): Reserve transfer to Trappist")
+    .finalizedEvents()
+    .run()
+
+  for (const { event } of events) {
+    if (RuntimeEvent.isXcmpQueue(event) && isXcmpMessageSent(event.value)) {
+      console.log("XcmpMessageSent.messageHash:", event.value)
+    }
+  }
+}
+
+// Retrieve billy's balance on Trappist.
+const { balance: billyTrappistBalance } = await retry(
+  () =>
+    Trappist.Assets.Account.value([TRAPPIST_ASSET_ID, billy.publicKey]).unhandle(undefined).run(),
+  retryOptions,
+)
+
+// Ensure the balance is greater than zero.
+console.log("Trappist(Billy): asset balance:", billyTrappistBalance)
+assert(billyTrappistBalance > 0)
+
+// Retrieve Billy's balance on statemint.
+const billyStatemintBalanceFinal = await billyStatemintBalance.run()
+
+// Ensure the balance is different from the initial.
+console.log("Statemine(Billy): asset balance:", billyStatemintBalanceFinal)
+assertNotEquals(billyStatemintBalanceInitial, billyStatemintBalanceFinal)
+
+// Retrieve the statemint sovereign account balance.
+const statemintSovereignAccountBalance = await Statemine.Assets.Account
+  // Sovereign address on sibling chain
```

Comment:
We can probably get rid of these two lines of comments. As well as have `214-217` on a single line.

Reply:
... unless, do you think it's important to describe this codec usage?

Reply:
you are right, these comments are not necessary